### PR TITLE
Add features to the s3 driver

### PR DESCRIPTION
**Pull request by AndreyKostov**
Do not merge yet, as this is blocked on the v4 auth option.

### CODE REVIEW

**Comment by AndreyKostov** [comment]
This addresses #30 #32 #33

**Comment by dmp42** [comment]
Thanks a lot @AndreyKostov !
cc @BrianBland can you do a first review?

File: storagedriver/s3/s3_test.go [hunk 1]
```diff
@@ -28,11 +30,42 @@ func init() {
 	}
 
 	s3DriverConstructor := func(region aws.Region) (storagedriver.StorageDriver, error) {
-		shouldEncrypt, err := strconv.ParseBool(encrypt)
-		if err != nil {
-			return nil, err
+		encryptBool := true
```
Comment (BrianBland) [inline storagedriver/s3/s3_test.go]: `encrypt` should default to `false` like the semantic defined in the README. Also, you should remove this from the list of environment variables required to run the tests.

File: storagedriver/s3/s3_test.go [hunk 2]
```diff
@@ -21,6 +21,8 @@ func init() {
 	secretKey := os.Getenv("AWS_SECRET_KEY")
 	bucket := os.Getenv("S3_BUCKET")
 	encrypt := os.Getenv("S3_ENCRYPT")
+	secure := os.Getenv("S3_SECURE")
+	v4auth := os.Getenv("S3_USE_V4_AUTH")
 	region := os.Getenv("AWS_REGION")
 	root, err := ioutil.TempDir("", "driver-")
```
Comment (BrianBland) [inline storagedriver/s3/s3_test.go]: We should delete this tempdir or just use a different method for generating a random root directory.

**Comment by BrianBland** [comment]
@AndreyKostov github seems to be working correctly again, so try re-triggering the circle build.

**Comment by BrianBland** [comment]
LGTM

**Comment by stevvooe** [comment]
LGTM!

**Pull request merged**